File: hw1/program/tetrahedron.cpp
//------------------------------------------------------------------------------
// tetrahedron.cpp - содержит процедуры для работы с правильным тетраэдром
//------------------------------------------------------------------------------

#define _CRT_SECURE_NO_WARNINGS
#include "stdio.h"
#include "math.h"
#include "tetrahedron.h"

//------------------------------------------------------------------------------
// Ввод параметров тетраэдра из файла
void In(tetrahedron& t, FILE* file) {
    fscanf(file, "%d", &t.a);
}

// Случайный ввод параметров тетраэдра
void InRnd(tetrahedron& t) {
    t.a = Random();
}

//------------------------------------------------------------------------------
// Вывод параметров тетраэдра в файл
void Out(tetrahedron& t, FILE* file) {
    fprintf(file, "It is Regular Tetrahedron. Edge length: a = %d. Surface area = %f. Material dencity: ",
        t.a, SurfaceArea(t));
}

//------------------------------------------------------------------------------
// Вычисление площади поверхности тетраэдра
double SurfaceArea(tetrahedron& t) {
    return double(sqrt(3) * t.a * t.a);
}

File: hw3/program/tetrahedron.py
import math
import random
from shape import *
# ----------------------------------------------


class Tetrahedron(Shape):
    def __init__(self):
        self.a = 0
        self.material_density = 0.0

    def get_figure_from_file(self, data, i):
        self.material_density = float(data[i])
        self.a = int(data[i + 1])

    def get_random_figure(self):
        self.a = random.randint(1, 101)
        self.material_density = random.uniform(1, 10001)

    def surface_area(self):
        return math.sqrt(3) * math.pow(self.a, 2)

    def write(self, output):
        output.write("It is Regular Tetrahedron: Edge length: a = {}. "
                     "Surface area = {}. Material density = {}"
                     .format(self.a, self.surface_area(), self.material_density))
